{
  "_guid_": "S-1009.4",
  "_extends_": "F-1009_Test_Data_Base_Error_Response",

  "title": "CaseLink field contains Invalid CaseReference value and Submit Case Creation Event is invoked on v1_external#/case-details-endpoint/saveCaseDetailsForCaseWorkerUsingPOST",

  "prerequisites" : [ {
      "Token_Creation": "S-1009_GetCreateToken"
    }
  ],

  "specs": [
    "an active profile in CCD",
    "contains some invalid case data",
    "has the 422 return code",
    "contains correctly configured CaseLink field with valid Case Reference"
  ],

  "request": {
    "body": {
      "data": {
        "CaseLinkField": {
          "CaseReference": "${}${[scenarioContext][childContexts][F-1009_CreateCasePreRequisiteCaseworkerBase][testData][actualResponse][body][id]}"
        },
        "TextField" : 3
      },
      "event": {
        "id": "createCaseCaseLinks",
        "summary": "",
        "description": ""
      },
      "security_classification": "PUBLIC",
      "event_token" : "${[scenarioContext][childContexts][Token_Creation][testData][actualResponse][body][token]}",
      "ignore_warning": true
    }
  },

  "expectedResponse": {
    "_extends_": "Common_422_Response",
    "body": {
      "exception" : "uk.gov.hmcts.ccd.endpoint.exceptions.CaseValidationException",
      "error" : "Unprocessable Entity",
      "message" : "Case data validation failed",
      "path" : "[[ANY_STRING_NOT_NULLABLE]]",
      "details" : {
        "field_errors": [
          {
            "id" : "TextField",
            "message" : "number is not a string"
          }
        ]
      },
      "callbackErrors" : null,
      "callbackWarnings" : null
    }
  }
}
